utils/docker/Dockerfile.centos.8

```diff
@@ -9,31 +9,63 @@
 FROM centos:8
 LABEL maintainer="daos@daos.groups.io"
 
+ARG REPO_URL
+ARG REPO_DAOS=""
+ARG REPO_DISTRO=""
+
 # Intermittent cache-bust.  Used to reduce load on the actual CACHEBUST later.
 ARG CB0
 
-ARG REPO_URL
-# Need to clean this up later
-# This is largely the same as the centos.7 code, however this sets priority for
-# the new repo, and leaves the default ones enabled as there is some metadata
-# that is not available locally.
-RUN if [ -n "$REPO_URL" ]; then						                           \
-      version=$(sed -e 's/.* \([0-9\.]*\)\.[0-9]*.*/\1/' /etc/centos-release); \
-      MY_REPO="${REPO_URL}repository/centos-${version}-x86_64-group/";         \
-      MY_NAME="${MY_REPO#*//}";						                           \
-      MY_NAME="${MY_NAME//\//_}";					                           \
+# If a local distro repository is supplied, then we should use it
+# and disable using the distro repositories that can be mirrored.
+# The Docker image starts out with the DISTRO provided GPG keys
+# on the disk, but not installed.  These must be isntalled before
+# adding any repo that needs GPG keys but does not provide meta
+# data for installing those keys.  Then the epel-release
+# package must be installed and the GPG key it provides must be
+# installed.
+RUN if [ -n "$REPO_DISTRO" ]; then                              \
+      rpm --import /etc/pki/rpm-gpg/*;                          \
+      MY_REPO="${REPO_URL}${REPO_DISTRO}/";                     \
+      MY_NAME="${MY_REPO#*//}";                                 \
+      MY_NAME="${MY_NAME//\//_}";                               \
       echo -e "[${MY_NAME}]\n\
 name=created from ${MY_REPO}\n\
 baseurl=${MY_REPO}\n\
 enabled=1\n\
-priority=1\n\
-gpgcheck=False\n" >> /etc/yum.repos.d/local-centos-group.repo;                 \
+repo_gpgcheck=0\n\
+gpgcheck=1\n" >> /etc/yum.repos.d/local-centos-group.repo;      \
+      dnf clean all;                                            \
+      dnf list --showduplicates epel-release dnf-plugins-core dnf-utils; \
+      dnf -y install --disablerepo extras --disablerepo baseos  \
+          epel-release-8-8.el8 dnf-plugins-core dnf-utils;      \
+      rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-8;         \
+      dnf config-manager --assumeyes --quiet --disable          \
+          baseos powertools epel epel-modular extras;           \
+      sed "s/^mirrorlist_expire=0*/mirrorlist_expire=99999999/" \
+          -i /etc/dnf/dnf.conf;                                 \
+      dnf -y update epel-release;                               \
+      dnf clean all;                                            \
```

Comment:
I'm really confused by this, you install a specific version of epel-release, disable epel and then update epel-release?  I can't work out what the purpose of that is at all.

Reply:
But if the epel repo is already enabled why do we need to install epel-release?

Reply:
It's enabled, but it's key is not installed.  Installing epel-release from CentOS _extras_ installs the key so that future installs from epel, signed by epel, will validate the signature checking.

Reply:
Where is it enabled, or does it all come from REPO_DISTRO?

Reply:
Yes, `REPO_DISTRO` is something Nexus calls a group repo in which it proxies a single repo to any number of other repos, providing a "combined" view of multiple individual repos.

Reply:
Ah, that's a new one on me.  I'll take another look tomorrow.

I assume I can see the values passed to REPO_DISTRO in CI and test with the same values locally?